chopper.py

```diff
@@ -119,62 +118,85 @@ def parse_args() -> dict[str, Any]:
         default=0,
     )
 
-    args = vars(parser.parse_args())
+    config = parser.parse_args()
 
-    if not args["input_path"].exists():
-        parser.error(
-            "Invalid input_path. Please specify an existing file or directory."
-        )
+    for path in config.input_paths:
+        if not path.exists():
+            parser.error(
+                "Invalid input_paths. Please specify existing file(s) and/or directory (directories)."
+            )
 
-    if not args["output_directory"].exists():
-        os.makedirs(args["output_directory"], exist_ok=True)
+    if not config.output_directory.exists():
+        config.output_directory.mkdir(exist_ok=True)
 
-    actions = ["columns", "rows", "equal", "shuffles"]
-    if not args.keys() & actions:
+    actions = {"columns", "rows", "equal", "shuffles"}
+    if len(set(config.__dict__.keys()).intersection(actions)) == 0:
         parser.error(
             "You must specify at least one action (columns, rows, equal, shuffles)."
         )
 
-    return args
+    return config
+
 
+def get_offsets(file: TextIOWrapper) -> list[int]:
+    pos = 0
+    row_offsets = []
 
-def shuffle_file(filepath: Path, shuffles: int, config: Config) -> list[str]:
+    for line in file:
+        pos += len(line)
+        row_offsets.append(pos)
+
+    return row_offsets
+
+
+def shuffle_files(
+    filepath: Path, shuffles: int, config: argparse.Namespace
+) -> list[str]:
 
     """Creates N copies of the provided file with the headers intact and data lines
     shuffled.
 
         Args:
             filepath (Path): The path of the file to shuffle.
             shuffles (int): Number of shuffles to perform. Outputs one file per shuffle.
-            config (Config): Config object created from CLI arguments.
+            config (Namespace): Namespace object created from CLI arguments.
 
         Returns:
             list[Path]: List of shuffled intermediate files.
     """
 
     files = []
-    with open(filepath, "r", encoding=config.encoding) as fin:
-        rows = fin.readlines()
 
-    headers = rows[0]
-    data = rows[1:]
+    with filepath.open("r", encoding=config.encoding, newline="") as fin_obj:
+        offsets = get_offsets(fin_obj)
+
+        with mmap.mmap(fin_obj.fileno(), length=0, access=mmap.ACCESS_READ) as fin:
+            if hasattr(mmap.mmap, "madvise"):
+                fin.madvise(mmap.madvise(mmap.MADV_RANDOM))
```

Comment:
I actually don't have any native Linux/MacOS devices at the moment and while Python running on WSL does have `madvise()`, I doubt it actually does anything. If someone with a Mac wants to drag race, that'd be cool. I can't really imagine this hurting too much, but you never know.